### PR TITLE
Add systemd-resolved support

### PR DESCRIPTION
**Pull request by tiran**
Fedora 33 switched to systemd-resolved

- [X] Add helpers to get forwarders from resolve1 D-BUS API
- [X] Configure NetworkManager to use systemd-resolved
- [X] Use new API for auto-forwarders
- [X] Configure systemd-resolved to use IPA's BIND server
- [X] Update DNS resolver configuration in ``ipa-server-upgrade``

See: https://pagure.io/freeipa/issue/8275

### CODE REVIEW

File: ipalib/install/dnsforwarders.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import os
+import socket
+
+import dbus
+
+from ipaplatform.paths import paths
+try:
+    from ipapython.dnsutil import get_ipa_resolver
+except ImportError:
+    from dns.resolver import get_default_resolver as get_ipa_resolver
```
Comment (tiran) [inline ipalib/install/dnsforwarders.py:17]: I'll remove this try/except block later. It makes it easier to test new code with current IPA version on F33.

**Comment by tiran** [comment]
I have successfully tested the PR on our test infrastructure:

```
# resolvectl 
Global
       LLMNR setting: resolve             
MulticastDNS setting: no                  
  DNSOverTLS setting: no                  
      DNSSEC setting: no                  
    DNSSEC supported: no                  
Fallback DNS Servers: 1.1.1.1             
                      8.8.8.8             
                      1.0.0.1             
                      8.8.4.4             
                      2606:4700:4700::1111
                      2001:4860:4860::8888
                      2606:4700:4700::1001
                      2001:4860:4860::8844

Link 2 (eth0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
DefaultRoute setting: yes                      
       LLMNR setting: yes                      
MulticastDNS setting: no                       
  DNSOverTLS setting: no                       
      DNSSEC setting: no                       
    DNSSEC supported: no                       
         DNS Servers: 10.11.5.19               
                      10.5.30.45               
          DNS Domain: ~.                       
                      openstacklocal
```

```
# ipa-server-install -p Secret123 -a Secret123 -r IPA.EXAMPLE -n ipa.example --no-ntp --setup-dns --auto-forwarders -U
The log file for this installation can be found in /var/log/ipaserver-install.log
==============================================================================
This program will set up the FreeIPA Server.
Version 4.9.0.dev202009221449+git8f034c75d4

This includes:
  * Configure a stand-alone CA (dogtag) for certificate management
  * Create and configure an instance of Directory Server
  * Create and configure a Kerberos Key Distribution Center (KDC)
  * Configure Apache (httpd)
  * Configure DNS (bind)
  * Configure the KDC to enable PKINIT

Excluded by options:
  * Configure the NTP client (chronyd)

Warning: skipping DNS resolution of host host-10-0-139-6.ipa.example
Checking DNS domain ipa.example., please wait ...
Checking DNS forwarders, please wait ...

The IPA Master Server will be configured with:
Hostname:       host-10-0-139-6.ipa.example
IP address(es): 10.0.139.6
Domain name:    ipa.example
Realm name:     IPA.EXAMPLE

The CA will be configured with:
Subject DN:   CN=Certificate Authority,O=IPA.EXAMPLE
Subject base: O=IPA.EXAMPLE
Chaining:     self-signed

BIND DNS server will be configured to serve IPA domain with:
Forwarders:       10.11.5.19, 10.5.30.45
Forward policy:   only
Reverse zone(s):  No reverse zone
...
The ipa-server-install command was successful
```

```
# resolvectl 
Global
       LLMNR setting: resolve             
MulticastDNS setting: no                  
  DNSOverTLS setting: no                  
      DNSSEC setting: no                  
    DNSSEC supported: no                  
  Current DNS Server: 127.0.0.1           
         DNS Servers: 127.0.0.1           
Fallback DNS Servers: 1.1.1.1             
                      8.8.8.8             
                      1.0.0.1             
                      8.8.4.4             
                      2606:4700:4700::1111
                      2001:4860:4860::8888
                      2606:4700:4700::1001
                      2001:4860:4860::8844
          DNS Domain: ~.                  
                      ipa.example         

Link 2 (eth0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
DefaultRoute setting: yes                      
       LLMNR setting: yes                      
MulticastDNS setting: no                       
  DNSOverTLS setting: no                       
      DNSSEC setting: no                       
    DNSSEC supported: no                       
  Current DNS Server: 10.11.5.19               
         DNS Servers: 10.11.5.19               
                      10.5.30.45               
          DNS Domain: ~.                       
                      openstacklocal
```

dig uses systemd-resolved stub resolver and gives correct answer:

```
# dig -t SRV _ldap._tcp.ipa.example

; <<>> DiG 9.11.22-RedHat-9.11.22-1.fc33 <<>> -t SRV _ldap._tcp.ipa.example
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 62959
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;_ldap._tcp.ipa.example.                IN      SRV

;; ANSWER SECTION:
_ldap._tcp.ipa.example. 86400   IN      SRV     0 100 389 host-10-0-139-6.ipa.example.

;; Query time: 1 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Tue Sep 22 11:02:28 EDT 2020
;; MSG SIZE  rcvd: 98
```

**Comment by tiran** [comment]
```
# cat /etc/resolv.conf
# This file is managed by man:systemd-resolved(8). Do not edit.
...
nameserver 127.0.0.53
options edns0 trust-ad
search ipa.example openstacklocal
```

**Comment by abbra** [comment]
LGTM. Let's merge this part and see what needs to be done with DNSSEC.

**Comment by tiran** [comment]
master:

* 96edff0b8c526763d75dc45d26841dc2cd1a70cf Add helpers for resolve1 and nameservers
* e64f27fdf8970c22052ec010f3b7bada084c743b Configure NetworkManager to use systemd-resolved
* 528c519cb5a30df691f00edba6ef6f3a6ab7df56 Use new API for auto-forwarders
* d12f1b4b39ef514c2bcb75b55c3f300767aca64d Configure systemd-resolved to use IPA's BIND
* 79b9982b86b68c708275afaaf1927c24db7c2eb1 Create systemd-resolved configuration on update

**Pull request closed without merging**